async-nats/src/connector.rs

```diff
@@ -59,6 +61,7 @@ pub(crate) struct ConnectorOptions {
     pub(crate) retain_servers_order: bool,
     pub(crate) read_buffer_capacity: u16,
     pub(crate) reconnect_delay_callback: Box<dyn Fn(usize) -> Duration + Send + Sync + 'static>,
+    pub(crate) auth_callback: Option<CallbackArg1<Vec<u8>, Result<Auth, AuthError>>>,
```

Comment:
Can't pass as a slice?

```suggestion
    pub(crate) auth_callback: Option<CallbackArg1<Vec<u8>, Result<Auth, AuthError>>>,
```

Reply:
I prefer to own things over async callback.
Especially that this is called only once per reconnect, so impact is irrelevant.